tests/basic/while.sp

```diff
@@ -10,4 +10,31 @@ public main()
     if (sFirst > 0 && sLast > 0 && donothing())
       break;
   }
+
+  int i;
+
+  i = 0;
+  do {
+    if (i % 2 == 0) {
+      if (i >= 1) {
+        print("break\n");
+        break;
+      }
+      else {
+        print("even\n");
+      }
+
+      continue;
+    }
+
+    print("old\n");
```

Comment:
Should this say odd?

Reply:
I'm holding off merging until I can run it against the corpus, which will probably be a day or two.

If PerfectLaugh wants to fix the typo before then, it'd be good to also split these two new cases into separate test files. Debugging tests is much easier when they're small, so we try not to make a single test function too big.

But I'm okay to merge this either way, typo fixed and tests split or not, I can fix that later.

Reply:
Sorry my English is bad ;\

@dvander I migrated the do-while part into separate file named "do-while.sp"